### ISSUE TITLE
scraping metrics is slow

### ISSUE THREAD

**Issue by johnbent** [Seagate/cortx#572]
For queries to github which take a since field, start using that field and using pickles to save old stuff.
For example, change repo->get_commits() to:

cortx_community.get_commits(ps,repo):
   url = repo.get_url()
   (latest,items)=ps.get_items(url)
   new_items = repo.get_commits(since=latest)
   for item in new_items:
       latest = max(latest,item.timestamp)
   ps.add_items(url,latest,new_items|items)

ps.add_items(key,ts,items):
    items_structure[key]=(latest,items)
    persist(items_structure)

ps.get_items(key):
   return items_structure[key]

And create a new items pickle and hope it doesn't grow too big.  If it is too big, then will need to split into multiple pickles.  One for commits, one for comments, one for pull requests, etc.

**Comment by mukul-seagate11** [comment]
@johnbent, to whom this issues needs to be assigned or direct to as I am unclear?

**Comment by johnbent** [comment]
I assigned it to myself.  Thanks.

**Comment by johnbent** [comment]
I might just close this one.  The metrics somehow broke in a very strange way.  I honestly suspect that github changed something on their back-end.  In the pygithub api, I used to do this to check the rate:
 gh.rate_limiting
And it worked!  But just yesterday, it stopped working.  It would query once and then cache the answer.  So I'd keep checking it and it would also look like the rate limiting was fine until the scripts would due to rate limiting.  So weird that it's behavior changed.  I didn't change anything in my scripts nor the version of python nor the modules.  I think that it must query a 'can-cache' value or something which changed server side.

Anyway, it took forever to debug but finally I realized it's behavior had changed and I replaced it with:
gh.get_rate_limit() 
which works as expected.

So the scripts are slow but since they run only once a week in the background, I don't care that much.  When they were failing, I was thinking we needed another solution.

So let's leave this open for a few weeks and if the scripts are fine, we can close it.
